store/precomputed_key/weaveVM/rpc/wvm_rpc_client.go

```diff
@@ -0,0 +1,241 @@
+	return rpc.signer.SignTransaction(ctx, &signData)
+}
+
+func (rpc *RPCClient) sendRawTransaction(ctx context.Context, signedTxHex string) (string, error) {
+	var err error
+	var signedTxBytes []byte
+
+	if strings.HasPrefix(signedTxHex, "0x") {
+		signedTxBytes, err = hexutil.Decode(signedTxHex)
+		if err != nil {
+			return "", fmt.Errorf("failed to decode signed transaction: %w", err)
+		}
+	} else {
+		signedTxBytes, err = hex.DecodeString(signedTxHex)
+		if err != nil {
+			return "", fmt.Errorf("failed to decode signed transaction: %w", err)
+		}
+	}
+
+	tx := new(ethtypes.Transaction)
+	err = tx.UnmarshalBinary(signedTxBytes)
+	if err != nil {
+		err = rlp.DecodeBytes(signedTxBytes, tx)
+		if err != nil {
+			return "", fmt.Errorf("failed to parse signed transaction: %w", err)
+		}
+	}
+
+	err = rpc.client.SendTransaction(ctx, tx)
+	if err != nil {
+		return "", err
+	}
+
+	rpc.log.Info("weaveVM: successfully sent transaction", "tx hash", tx.Hash().String())
+
+	err = rpc.logReceipt(tx)
+	if err != nil {
+		rpc.log.Error("failed to log sent transaction receipt", "error", err)
+	}
+
+	return tx.Hash().String(), nil
+}
+
+func (rpc *RPCClient) logReceipt(tx *ethtypes.Transaction) error {
+	var txDetails Transaction
+	txBytes, err := tx.MarshalJSON()
+	if err != nil {
+		return err
+	}
+	if err := json.Unmarshal(txBytes, &txDetails); err != nil {
+		return err
+	}
+
+	txDetails.TransactionTime = tx.Time().Format(time.RFC822)
+	txDetails.TransactionCost = tx.Cost().String()
+
+	convertFields := []string{"Nonce", "MaxPriorityFeePerGas", "MaxFeePerGas", "Value", "Type", "Gas"}
+	for _, field := range convertFields {
+		if err := convertHexField(&txDetails, field); err != nil {
+			return err
+		}
+	}
+
+	txJSON, err := json.MarshalIndent(txDetails, "", "\t")
+	if err != nil {
+		return err
+	}
+
+	rpc.log.Info("weaveVM: transaction receipt", "tx receipt", string(txJSON))
+	return nil
+}
+
+// Transaction represents the structure of the transaction JSON.
+type Transaction struct {
+	Type                 string   `json:"type"`
+	ChainID              string   `json:"chainId"`
+	Nonce                string   `json:"nonce"`
+	To                   string   `json:"to"`
+	Gas                  string   `json:"gas"`
+	GasPrice             string   `json:"gasPrice,omitempty"`
+	MaxPriorityFeePerGas string   `json:"maxPriorityFeePerGas"`
+	MaxFeePerGas         string   `json:"maxFeePerGas"`
+	Value                string   `json:"value"`
+	Input                string   `json:"input"`
+	AccessList           []string `json:"accessList"`
+	V                    string   `json:"v"`
+	R                    string   `json:"r"`
+	S                    string   `json:"s"`
+	YParity              string   `json:"yParity"`
+	Hash                 string   `json:"hash"`
+	TransactionTime      string   `json:"transactionTime,omitempty"`
+	TransactionCost      string   `json:"transactionCost,omitempty"`
+}
+
```

Comment:
can we please change the package name to snake-case `weave_vm` to ensure compatibility with existing module naming schema?

Reply:
renamed package weavevm => weave_vm